esda/moran.py

```diff
@@ -212,9 +231,9 @@ def __moments(self):
         self.EI = -1.0 / (self.n - 1)
         n = self.n
         n2 = n * n
-        s1 = self.w.s1
-        s0 = self.w.s0
-        s2 = self.w.s2
+        s1 = self.w._s1 if hasattr(self.w, "_s1") else self.w.s1
```

Comment:
This seems like a less elegant approach then having s0, s1, s2 be public on the graph? The latter would avoid the conditional (I think)?

Reply:
It is byt of we want to deprecate public properties than it is a way forward. As I said in the libpysal PR I don't mind which way we go.